src/main/java/frc/robot/states/ShootState.java

```diff
@@ -0,0 +1,38 @@
+package frc.robot.states;
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import frc.robot.Constants;
+import frc.robot.swerve.SwerveDrive;
+import lib.Utils;
+
+public class ShootState implements ScoreState {
+
+    @Override
+    public Command driveToClosestOptimalPoint() {
+        return Commands.defer(
+                () -> {
+                    var optimalPoints = Constants.OPTIMAL_POINTS_SHOOT_BLUE;
+                    if (isRed()) {
+                        optimalPoints = Constants.OPTIMAL_POINTS_SHOOT_RED;
+                    }
+                    Pose2d optimalPose =
+                            Utils.calcOptimalPose(
```

Comment:
You could probably use [`Pose2d.nearest()`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/math/geometry/Pose2d.html#nearest(java.util.List)) (there is also a [`Translation2d.nearest()`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/math/geometry/Translation2d.html#nearest(java.util.List))).

Reply:
39c9fb6